packages/suite/src/views/wallet/coinmarket/exchange/detail/index.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { AppState } from '@suite-types';
+import {
+    useCoinmarketExchangeDetail,
+    CoinmarketExchangeDetailContext,
+} from '@wallet-hooks/useCoinmarketExchangeDetail';
+import { WalletLayout } from '@wallet-components';
+import styled from 'styled-components';
+import { ComponentProps, Props } from '@wallet-types/coinmarketExchangeDetail';
+import { connect } from 'react-redux';
+
+import Detail from './Detail';
+
+const mapStateToProps = (state: AppState): ComponentProps => ({
+    selectedAccount: state.wallet.selectedAccount,
+    trades: state.wallet.coinmarket.trades,
+    transactionId: state.wallet.coinmarket.exchange.transactionId,
+});
+
+const Wrapper = styled.div`
+    display: flex;
+    width: 100%;
+    flex-direction: column;
+`;
+
+const DetailIndexLoaded = (props: Props) => {
+    const { selectedAccount } = props;
+    const coinmarketExchangeContextValues = useCoinmarketExchangeDetail({
+        ...props,
+        selectedAccount,
+    });
+
+    return (
+        <CoinmarketExchangeDetailContext.Provider value={coinmarketExchangeContextValues}>
+            <Wrapper>
+                <Detail />
+            </Wrapper>
+        </CoinmarketExchangeDetailContext.Provider>
+    );
+};
+
+const DetailIndex = (props: ComponentProps) => {
+    const { selectedAccount } = props;
+
+    if (selectedAccount.status !== 'loaded') {
+        return <WalletLayout title="TR_NAV_EXCHANGE" account={selectedAccount} />;
+    }
+    return <DetailIndexLoaded {...props} selectedAccount={selectedAccount} />;
+};
+
+export default connect(mapStateToProps)(DetailIndex);
```

Comment:
lets use hooks instead of connected component from now on, like it was [changed here](https://github.com/trezor/trezor-suite/commit/2fa02c3e3711c6c9c1cb9019d784eb4cf5d86942)

you don't have to introduce children props, but it would be useful to write hooks tests

Reply:
Done, for now without children